.gitlab-ci.yml

```diff
@@ -0,0 +1,132 @@
+# see https://docs.gitlab.com/ce/ci/yaml/README.html for all available options
+
+variables:
+  SCHEDULER_PARAMETERS: "-J ArborX_CI -W 0:30 -nnodes 1 -P CSC333 -alloc_flags smt1"
+
+stages:
+  - buildDependencies
+  - buildArborX
+  - runBenchmarks
+  - compare
+
+.LoadModules:
+  before_script:
+    - module load gcc/7.4.0 cuda/10.1.243 cmake/3.15.2 git/2.20.1 spectrum-mpi/10.3.1.2-20200121
+
+.BuildBoost:
+  extends: .LoadModules
+  stage: buildDependencies
+  script:
+    - BOOST_VERSION=1.67.0 &&
+      BOOST_VERSION_UNDERSCORE=$(echo "$BOOST_VERSION" | sed -e "s/\./_/g") &&
+      BOOST_URL=https://dl.bintray.com/boostorg/release/${BOOST_VERSION}/source &&
+      BOOST_ARCHIVE=boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 &&
+      wget --quiet ${BOOST_URL}/${BOOST_ARCHIVE} &&
+      mkdir -p boost &&
+      tar -xf ${BOOST_ARCHIVE} -C boost --strip-components=1 &&
+      cd boost &&
+      ./bootstrap.sh --prefix=/ccsopen/proj/csc333/boost.install &&
+      ./b2 -j8 variant=release cxxflags=-w install
+  tags:
+    - nobatch
+
+.BuildBenchmark:
+  extends: .LoadModules
+  stage: buildDependencies
+  script:
+    - git clone https://github.com/google/benchmark.git -b v1.4.1 &&
+      cd benchmark &&
+      git clone https://github.com/google/googletest.git -b release-1.8.1 &&
```

Comment:
cf #312

Reply:
Currently, we are not using this anyway but use a built library. I agree that we should keep this in sync, though. If necessary I can do that here or we add it to the other pull request if we decide to merge this first.

Reply:
Just do it here, and we'll merge immediately after.